lib/container/ModuleFederationPlugin.js

```diff
@@ -87,6 +122,7 @@ class ModuleFederationPlugin {
 					shareScope: options.shareScope
 				}).apply(compiler);
 			}
+			new HoistContainerReferences().apply(compiler);
```

Comment:
If I understand correctly we do this only for `References`? Perhaps for performance purposes we should add a check:
```
options.remotes &&
				(Array.isArray(options.remotes)
					? options.remotes.length > 0
					: Object.keys(options.remotes).length > 0)
```

like we do for `ContainerReferencePlugin`?

Reply:
I dont think that will work without introducing other problems with eager modules. 

think of this case: 
User shares modules with eager:true, shared modules will only hoist to entrypoint, not runtime - if runtime eagerly share and someone eager require the share - the host runtime will not have the external reference (if the share depends on a external) 

So plugin maybe can be renamed, since its actually HoistEagerExternals. 
In v2, reference hoisting also is responsible for hoisting `@module-federation/runtime,webpack-bundler-runtime,sdk` - since splitChunks will also interfere with federation - this should prevent split chunks from moving eager modules into another chunk beside the runtime.

So while i use it for other things as well, there are cases in federation where it will still fail as anything the system uses or may provide eagerly needs to be in the root chunk at startup, which can be external modules of a shared package as well. 

Id say the only time we dont need to apply it is when there are no seperate `runtimeChunk` added. So like if its just main.js, then webpack already hoists the module to entry chunk. But when there is runtimeChunk, it is only added later in compile, so all modules "hoist" to entry chunk only, not root runtime chunk .